vmm/src/lib.rs

```diff
@@ -1095,23 +1095,43 @@ impl Vmm {
             arch::HIMEM_START,
         )
         .map_err(StartMicrovmError::KernelLoader)?;
+
+        // This is x86_64 specific since on aarc64 the commandline will be specified through the FDT.
+        #[cfg(target_arch = "x86_64")]
         kernel_loader::load_cmdline(vm_memory, kernel_config.cmdline_addr, &cmdline_cstring)
             .map_err(StartMicrovmError::LoadCommandline)?;
 
+        Ok(entry_addr)
+    }
+
+    fn configure_system(&mut self) -> std::result::Result<(), StartMicrovmError> {
+        let kernel_config = self
+            .kernel_config
+            .as_mut()
+            .ok_or(StartMicrovmError::MissingKernelConfig)?;
+        let cmdline_cstring = CString::new(kernel_config.cmdline.clone()).map_err(|_| {
```

Comment:
`cmdline_string` is now initialized both in `load_kernel` and `configure_system`. Does it's value change between the two invocations? If not, can we pull it out of the functions and pass it as a reference?